File: 4. Median of Two Sorted Arrays/main.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findMedianSortedArrays = function (nums1, nums2) {
  let merged = [];
  let i = 0,
    j = 0,
    m = nums1.length,
    n = nums2.length;
  let length = m + n;
  while (i < m && j < n) {
    if (nums1[i] < nums2[j]) merged.push(nums1[i++]);
    else merged.push(nums2[j++]);
  }

  while (i < m) merged.push(nums1[i++]);
  while (j < n) merged.push(nums2[j++]);

  if (length % 2 == 1) return merged[Math.floor(length / 2)];

  let Mid = length / 2;
  return (merged[Mid - 1] + merged[Mid]) / 2;
};

File: 4. Median of Two Sorted Arrays/main.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        
        m, n = len(nums1), len(nums2)
        halfLen = math.ceil((m + n)/2)
        if (m + n) % 2 == 0:
            halfLen += 1
        lp, rp = 0, 0
        arr = []

        for i in range(halfLen):
            if lp > m-1:
                arr.append(nums2[rp])
                rp += 1
                continue

            if rp > n-1:
                arr.append(nums1[lp])
                lp += 1
                continue
            
            if nums1[lp] <= nums2[rp]:
                arr.append(nums1[lp])
                lp += 1
            else: 
                arr.append(nums2[rp])
                rp += 1
        
        if (m + n) % 2 == 1:
            return arr[-1]

        return (arr[-1] + arr[-2]) / 2.0
                